File: go/queue/queue.go
package main

import "fmt"

type Node struct {
	value int
	next  *Node
}

var head *Node = nil
var tail *Node = nil
var length int = 0

func enqueue(value int) {
	var newNode *Node = &Node{value: value}
	if head == tail {
		head = newNode
		tail = newNode
		return
	}

	tail.next = newNode
	tail = newNode
	length += 1
}

func dequeue() int {
	if head == nil {
		return -1
	}

	var temp *Node = head
	value := temp.value

	if head == tail {
		head = nil
		tail = nil
	} else {
		head = head.next
	}

	temp = nil

	length -= 1
	return value
}

func main() {
	enqueue(5)
	fmt.Println(head.value)
}

File: c_c++/queue/queue.c
#include <stdio.h>
#include <stdlib.h>

struct Node
{
    int value;
    struct Node *next;
};

struct Node *head = NULL;
struct Node *tail = NULL;
int length = 0;

int peak()
{
    return head->value;
}

void enqueue(int value)
{
    struct Node *newNode = (struct Node *)malloc(sizeof(struct Node));
    newNode->value = value;
    newNode->next = NULL;

    if (head == tail)
    {
        head = tail = newNode;
        return;
    }

    tail->next = newNode;
    tail = newNode;
}

int dequeue()
{
    if (head == NULL)
    {
        return -1;
    }

    struct Node *temp = head;
    int value = temp->value;

    if (head == tail)
    {
        head = tail = NULL;
    }
    else
    {
        head = head->next;
    }

    free(temp);
    length--;

    return value;
};

int main()
{
    enqueue(5);
    printf("%d\n", head->value);
}